model-engine/model_engine_server/api/llms_v1.py

```diff
@@ -118,6 +119,9 @@ async def create_model_endpoint(
     """
     Creates an LLM endpoint for the current user.
     """
+    external_interfaces.monitoring_metrics_gateway.emit_route_call_metric(
```

Comment:
Should we emit a metric here? Or add tags to the current trace? cc @song-william

Reply:
I think adding tags to the current trace would "kill 2 birds with one stone" so we can search for traces based on user_id as well. 

Metrics generated by Traces are not sampled out: https://docs.datadoghq.com/tracing/metrics/metrics_namespace/

Reply:
@saiatmakuri If we add a tag to the existing trace, then add this logic into the `auth` dependency. This would then make it so we don't need to copy-pasta this call to each route.

Reply:
Ahh, it looks like we can't customize the tags for trace-based metrics. 

> Trace metrics tags, possible tags are: env, service, version, resource, sublayer_type, sublayer_service, http.status_code, http.status_class, Datadog Agent tags (including the host and second primary tag). Note: Tags set on spans do not count and will not be available as tags for your traces metrics.
https://docs.datadoghq.com/tracing/metrics/metrics_namespace/ (edited)